invenio_app_ils/search/permissions.py

```diff
@@ -98,6 +95,9 @@ def search_factory_filter_by_patron(self, search):
     """Prepare query string to filter records by current logged in user."""
     def filter_by_patron(query_string):
         """Filter search results by patron_pid."""
+        if not current_user.is_authenticated:
```

Comment:
I wanted to reuse `_ils_search_factory` but it always ensures that the user is authenticated which not all endpoints require. It made more sense to keep the `is_authenticated` check in `filter_by_patrons` since that's where it's applicable. @kprzerwa  Do you see any issues moving it there?